lib/blockchain.go

```diff
@@ -2617,23 +2630,39 @@ func (bc *Blockchain) DisconnectBlocksToHeight(blockHeight uint64) error {
 			}
 
 			// Set the best node hash to the new tip.
-			if err := PutBestHashWithTxn(txn, nil, &prevHash, ChainTypeDeSoBlock); err != nil {
-				return err
+			if bc.postgres != nil {
+				if err := bc.postgres.UpsertChain(MAIN_CHAIN, &prevHash); err != nil {
```

Comment:
it would be better if we could move this postgres v badger logic to the db adapter